### PR TITLE
Support Op Builders in Core Module

### PR DESCRIPTION
**Pull request by weilhuan-quic**
# WHAT
- Support Op implementations based on the wrappers.
- Use `TensorPool` to manage the intermediate tensors, `TensorPool` can be abstract or change to allocator if need.
- All builders are functions which receive `TensorPool`, input output `TensorWrapper`, and the op-specific parameters.
- All builders are functions which return `std::vector<OpWrapper>`
- The interface of all builders is open to change based on the requirement.
- Make these builders independent to LiteRT/tflite, use c++ primitive types as parameters.
- Only dependent on QNN and `Wrappers`.

# Supported Op 
| Supported Op | 
| --- | 
| Add |
| And |
| Cast  |
| Concat  |
| Cos |
| Div |
| Sin |
| Greater |
| Less |
| Mul |
| Rsqrt |
| Square |
| SquareDifference |
| Sub |
| EmbeddingLookUp |
| FullyConnected |
| Gather |
| GeLu |
| BatchMatMul |
| Mean |
| Quantize |
| Sum |
| Reshape |
| Select |
| SelectV2 |
| Slice |
| Softmax |
| Split |
| Tanh |
| Transpose |

### CODE REVIEW

**Comment by weilhuan-quic** [comment]
close because it is merged into tensorflow master.

**Comment by weilhuan-quic** [comment]
close because it is merged into tensorflow master.
